src/text-field.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const React = require('react');
+const mdl = require('material-design-lite/material.min');
+const classnames = require('classnames');
+
+const outerBaseClasses = {
+  'mdl-textfield': true,
+  'mdl-js-textfield': true
+};
+
+const inputBaseClasses = {
+  'mdl-textfield__input': true
+};
+
+const labelBaseClasses = {
+  'mdl-textfield__label': true
+};
+
+class Textfield extends React.Component {
+
+  componentDidMount(){
+    const node = React.findDOMNode(this);
+    mdl.upgradeElement(node, 'MaterialTextfield');
+  }
+
+  componentWillUnmount(){
+    const node = React.findDOMNode(this);
+    mdl.downgradeElements(node);
+  }
+
+  render(){
+    const {
+      className,
+      id,
+      type,
+      label,
+      floatingLabel,
+      error,
+      expandable,
+      button,
+      icon,
+      expandableHolder
+    } = this.props;
+
+    const outerClasses = classnames(outerBaseClasses, {
+      'mdl-textfield--floating-label': floatingLabel,
+      'mdl-textfield--expandable': expandable,
+      'mdl-button': button,
+      'mdl-js-button': button,
+      'mdl-button--icon': icon,
+      'mdl-textfield__expandable-holder': expandableHolder
+    }, className);
+
+    const inputClasses = classnames(inputBaseClasses);
+
+    const labelClasses = classnames(labelBaseClasses);
+
+    let textField, labelField = '', errorField = '';
+
+    if (type === 'textarea') {
+      textField = (<textarea className={inputClasses} id={id} />);
+    } else {
+      textField = (<input className={inputClasses} type={type} id={id} />);
+    }
+
+    if (label) {
+      labelField = (<label className={labelClasses} htmlFor={id}>{label}</label>);
+    }
+
+    if (error) {
+      errorField = (<span className='mdl-textfield__error'>{error}</span>);
+    }
+
+    return (
+      <div {...this.props} className={outerClasses}>
+        {textField}
+        {labelField}
+        {errorField}
+      </div>
+    );
+  }
+}
+
+Textfield.propTypes = {
+  className: React.PropTypes.string,
+  id: React.PropTypes.string.isRequired,
```

Comment:
could `id` be uniquely generated if not given?  seems weird to be required just to hook the label to the input